src/libs/tools/src/backendbuilder.cpp

```diff
@@ -409,6 +415,29 @@ void BackendBuilder::addPlugin (PluginSpec const & plugin)
 		newPlugin.appendConfig (provides.getConfig ());
 	}
 
+	// call plugin's checkconf function (if provided)
+	// this enables a plugin to verify its configuration at mount time
+	checkConfPtr checkConfFunction = nullptr;
+	try
+	{
+		checkConfFunction = reinterpret_cast<checkConfPtr> (pluginDatabase->getSymbol (newPlugin, "checkconf"));
+	}
+	catch (...)
+	{
+		// the checkconf operation is optional, so no worries if it was not found
+	}
+
+	if (checkConfFunction != nullptr)
+	{
+		ckdb::Key * errorKey = ckdb::keyNew (0);
+		int checkResult = checkConfFunction (errorKey, newPlugin.getConfig ().getKeySet ());
+		if (checkResult != 1)
```

Comment:
I would do `== -1` here. What about the convention:

- 0: config was not changed (was ok)
- 1: config is changed (now ok)
- -1: config not ok, could not fix.